cost-model/benches/block_cost_limit.rs

```diff
@@ -0,0 +1,136 @@
+#![feature(test)]
+extern crate test;
+use {
+    ahash::AHashMap,
+    std::collections::HashMap,
+    lazy_static::lazy_static,
+    solana_sdk::{
+
+        signer::Signer,
+    },
+    solana_sdk::{
+        address_lookup_table, bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable,
+        compute_budget, ed25519_program, loader_v4, pubkey::Pubkey, secp256k1_program,
+    },
+    test::Bencher,
+};
+
+struct BenchSetup {
+    pubkeys: [Pubkey; 12],
+    ahash_map: AHashMap<Pubkey, u64>,
+    hash_map : HashMap<Pubkey, u64>,
+}
+
+const NUM_TRANSACTIONS_PER_ITER: usize = 1024;
+const DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT: u32 = 200_000;
+
+fn setup() -> BenchSetup {
+    let pubkeys : [Pubkey; 12] = [
+        solana_stake_program::id(),
+        solana_config_program::id(),
+        solana_vote_program::id(),
+        solana_system_program::id(),
+        compute_budget::id(),
+        address_lookup_table::program::id(),
+        bpf_loader_upgradeable::id(),
+        bpf_loader_deprecated::id(),
+        bpf_loader::id(),
+        loader_v4::id(),
+        secp256k1_program::id(),
+        ed25519_program::id(),
+    ];
+
+    lazy_static! {
+    /// Number of compute units for each built-in programs
+    pub static ref ahash_map: AHashMap<Pubkey, u64> = [
+        (solana_stake_program::id(), solana_stake_program::stake_instruction::DEFAULT_COMPUTE_UNITS),
+        (solana_config_program::id(), solana_config_program::config_processor::DEFAULT_COMPUTE_UNITS),
+        (solana_vote_program::id(), solana_vote_program::vote_processor::DEFAULT_COMPUTE_UNITS),
+        (solana_system_program::id(), solana_system_program::system_processor::DEFAULT_COMPUTE_UNITS),
+        (compute_budget::id(), solana_compute_budget_program::DEFAULT_COMPUTE_UNITS),
+        (address_lookup_table::program::id(), solana_address_lookup_table_program::processor::DEFAULT_COMPUTE_UNITS),
+        (bpf_loader_upgradeable::id(), solana_bpf_loader_program::UPGRADEABLE_LOADER_COMPUTE_UNITS),
+        (bpf_loader_deprecated::id(), solana_bpf_loader_program::DEPRECATED_LOADER_COMPUTE_UNITS),
+        (bpf_loader::id(), solana_bpf_loader_program::DEFAULT_LOADER_COMPUTE_UNITS),
+        (loader_v4::id(), solana_loader_v4_program::DEFAULT_COMPUTE_UNITS),
+        // Note: These are precompile, run directly in bank during sanitizing;
+        (secp256k1_program::id(), 0),
+        (ed25519_program::id(), 0),
+    ]
+    .iter()
+    .cloned()
+    .collect();
+    }
+
+    lazy_static! {
+    /// Number of compute units for each built-in programs
+    pub static ref hash_map: AHashMap<Pubkey, u64> = [
```

Comment:
```suggestion
    pub static ref hash_map: HashMap<Pubkey, u64> = [
```

Reply:
To bench `BUILTIN_INSTRUCTION_COSTS`, you probably don't need to declare `ahash_map` nor `hash_map` here, instead just need one `bench_find()` that work against `BUILTIN_INSTRUCTION_COSTS`. So you run the bench when `BUILTIN_INSTRUCTION_COSTS` is hashmap, then run again after changingit to ahashmap, so we can compare the perf gain of the change.